translations/src/main/kotlin/com/hedvig/libs/translations/RemoteJsonFileClient.kt

```diff
@@ -0,0 +1,71 @@
+package com.hedvig.libs.translations
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.slf4j.LoggerFactory
+import java.net.URL
+import java.util.Timer
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.Future
+import java.util.concurrent.TimeUnit
+import kotlin.concurrent.timerTask
+
+class RemoteJsonFileClient(
+    private val url: String = "https://s3.eu-central-1.amazonaws.com/translations.hedvig.com/platform/all_locales.json",
+    refreshRateMinutes: Long = 10
+) : TranslationsClient {
+
+    private val logger = LoggerFactory.getLogger(RemoteJsonFileClient::class.java)
+    private val objectMapper = ObjectMapper()
+    private var translationsByLocale: JsonNode = objectMapper.readTree("{}")
+    private val initialRefreshFuture: Future<Unit> =
+        CompletableFuture.supplyAsync { translationsByLocale = refreshTranslations() }
+
+    init {
+        val refreshRateMs = TimeUnit.MINUTES.toMillis(refreshRateMinutes)
+
+        Timer().schedule(timerTask {
+            translationsByLocale = refreshTranslations()
+        }, refreshRateMs, refreshRateMs)
+    }
+
+    private fun refreshTranslations(): JsonNode {
+
+        try {
+            val startTime = System.currentTimeMillis()
+            val newTranslations = getRemoteFileAsJson(url)
+            logger.debug("Done refreshTranslations, took: ${System.currentTimeMillis() - startTime} ms")
+            return newTranslations
+        } catch (e: Exception) {
+            // Return the existing one
+            logger.error("Failed to download translations json file: ${e.message}", e)
```

Comment:
```suggestion
            logger.error("Failed to download translations json file.", e)
```
Adding the exception to the last parameter already makes it include the error message, so we don't need to inline it in our own message as well